test/metabase/api/action_test.clj

```diff
@@ -55,11 +56,22 @@
                                  :type "implicit"
                                  :model_id card-id
                                  :kind "row/create"
-                                 :parameters [{:id "x" :type "number"}]}]]
-            (doseq [initial-action posted-actions]
-              (mt/user-http-request :crowberto :post 200 "action" initial-action))
+                                 :parameters [{:id "x" :type "number"}]}
+                                {:name "Archived example"
+                                 :type "query"
+                                 :model_id card-id
+                                 :dataset_query (update (mt/native-query {:query "update venues set name = 'foo' where id = {{x}}"})
+                                                        :type name)
+                                 :database_id (mt/id)
+                                 :parameters [{:id "x" :type "number"}]
+                                 :visualization_settings {:position "top"}
+                                 :archived true}]]
+            (doseq [{archived? :archived :as initial-action} posted-actions]
+              (let [{action-id :id} (mt/user-http-request :crowberto :post 200 "action" (dissoc initial-action :archived))]
+                (when archived?
+                  (db/update! Action action-id :archived true))))
```

Comment:
This just seems a little strange... we're doing a POST and then updating the actions with `db/update!`, all to make an archived action.

This test shouldn't need to test the POST endpoint, since that should be covered by `unified-action-create-test`. I would have expected to see `mt/with-actions` being used here.

Reply:
definitely strange…I was just cargo-culting the existing test structure. Changed it to use `with-actions` and no HTTPing